### ISSUE TITLE
Configure a data type factory

### ISSUE THREAD

**Issue by H3n-42** [janbols/spock-dbunit#9]
When using an Oracle Database the following warning spams the log file:

`WARN org.dbunit.dataset.AbstractTableMetaData - Potential problem found: The configured data type factory 'class org.dbunit.dataset.datatype.DefaultDataTypeFactory' might cause problems with the current database 'Oracle' (e.g. some datatypes may not be supported properly). In rare cases you might see this message because the list of supported database products is incomplete (list=[derby]). If so please request a java-class update via the forums.If you are using your own IDataTypeFactory extending DefaultDataTypeFactory, ensure that you override getValidDbProducts() to specify the supported database products.`

It would be awesome when you could set the used database somehow

**Comment by janbols** [comment]
Hi @H3npi , normally you should be able to configure your `IDatabaseTester` by using the `configure` field in the `DbUnit` annotation like in the example below:
```
    @DbUnit(configure = { IDatabaseTester it ->
        it.connection.config.setProperty(DatabaseConfig.PROPERTY_DATATYPE_FACTORY, new OracleDataTypeFactory())
    })
    def content = {
        User(id: 1, name: 'xxx')
    }

```

See http://dbunit.sourceforge.net/faq.html#DefaultDataTypeFactory for more information.

Hope this helps.
Jan

**Comment by H3n-42** [comment]
Well i tried this before, and still got the message.
BR
H3npi

**Comment by janbols** [comment]
Hi @H3npi , I changed the way the `DatabaseTester` handles the `getConnection` method. The DbUnit one creates a new one for every call which seems plain wrong if you ask me. I overwrote that method.

Could you check if the latest commit https://github.com/janbols/spock-dbunit/commit/d568338ebe9ad495ff279b983b37ffc965163c03 solved your problem?

Thanks
Jan

**Comment by H3n-42** [comment]
Yes it does!

thank you!

**Comment by janbols** [comment]
The currently released version 0.4 contains the fixes for this bug. 
Thanks for the bug report.
